### PR TITLE
Feature/edit vod

### PR DESCRIPTION
**Pull request by mono424**
!! Should be merged after https://github.com/joschahenningsen/TUM-Live/pull/1027

### Motivation and Context
Currently a user who wants to update the VoD-Media, needs to delete the VoD and recreate it with updated Video files.

### Description
Allow a user to update VOD Media files.

### Steps for Testing

Prerequisites:
- 1 Lecturer
- 1 VoD

1. Log in
2. Navigate to a Course Admin Page
3. Click on edit on a VoD
4. Select new Video Files and Save VoD

### Screenshots
Bildschirmfoto 2023-05-29 um 17 03 08

### CODE REVIEW

**Comment by mono424** [comment]
Just as a note :)
Still some issues to fix:
- ERRO[04.06.2023 15:24:56] Error transcoding audio                       error="transcode stream audio: exit status 1: File '/Users/khadim/mass/2022/W/EAGAbschied/25311.m4a' already exists. Overwrite? [y/N] "

- Are the thumbnails being recreated ?

File: web/ts/edit-course.ts [hunk 1]
```diff
@@ -163,11 +166,15 @@ export class Lecture {
     }
 
     updateIsDirty() {
+        console.log(this.newCombinedVideo);
```
Comment (MatthiasReumann) [inline web/ts/edit-course.ts]:
```suggestion
```

File: api/stream.go [hunk 2]
```diff
@@ -138,7 +138,25 @@ func (r streamRoutes) getThumbs(c *gin.Context) {
...
+			c.AbortWithStatus(http.StatusNotFound)
+			return
+		}
+		c.File(thumb)
+		return
+	}
+
+	videoType := model.VideoType(queryType)
+	if !videoType.Valid() {
+		c.AbortWithStatus(http.StatusBadRequest)
```
Comment (MatthiasReumann) [inline api/stream.go]: ^ same here

File: api/stream.go [hunk 3]
```diff
@@ -138,7 +138,25 @@ func (r streamRoutes) getThumbs(c *gin.Context) {
 func (r streamRoutes) getVODThumbs(c *gin.Context) {
 	tumLiveContext := c.MustGet("TUMLiveContext").(tools.TUMLiveContext)
 
-	thumb, err := tumLiveContext.Stream.GetLGThumbnail()
+	queryType := c.Query("type")
+
+	if queryType == "" {
+		thumb, err := tumLiveContext.Stream.GetLGThumbnail()
+		if err != nil {
+			c.AbortWithStatus(http.StatusNotFound)
```
Comment (MatthiasReumann) [inline api/stream.go]:
```suggestion
			_ = c.Error(tools.RequestError{
	                          Status:        http.StatusNotFound,
	                          CustomMessage: "Large Thumbnail not found",
                        })
                        return
```

**Review by MatthiasReumann — CHANGES_REQUESTED** [review]
Works and looks great 🌟

A few nitpicks, but other than that I think this one is good to go.

**Review by MatthiasReumann — APPROVED** [review]
Thanks! 🚀

**Review by joschahenningsen — APPROVED** [review]
Nice!

**Pull request merged**